src/main/java/frc/robot/subsystems/climb/Climb.java

```diff
@@ -0,0 +1,220 @@
+        }
+    }
+
+    public void stop()
+    {
+        _io.setVoltageLeft(0.0);
+        _io.setVoltageRight(0.0);
+
+        _climbSetpointLeft  = null;
+        _climbSetpointRight = null;
+
+        setLockState(true);
+    }
+
+    public void setLockState(boolean enabled)
+    {
+        _io.setLockStateLeft(enabled, _inputs);
+        _io.setLockStateRight(enabled, _inputs);
+    }
+
+    public void setSetpointLeft(double setpoint)
+    {
+        _climbSetpointLeft = MathUtil.clamp(setpoint, _climbMinExtension, _climbMaxExtension);
+    }
+    
+    public void setSetpointRight(double setpoint)
+    {
+        _climbSetpointRight = MathUtil.clamp(setpoint, _climbMinExtension, _climbMaxExtension);
+    }
+
+    public void setVoltageLeft(double volts)
+    {
+        _climbSetpointLeft = null;
+        
+        _io.setLockStateLeft(Math.abs(volts) <= 0.12, _inputs); // TODO: Change number to constants
+        
+        _io.setVoltageLeft(volts);
+    }
+
+    public void setVoltageRight(double volts)
+    {
+        _climbSetpointRight = null;
+        
+        _io.setLockStateRight(Math.abs(volts) <= 0.12, _inputs); // TODO: Change number to constants
+        
+        _io.setVoltageRight(volts);
+    }
+    
+    public boolean isAtLeftSetpoint()
+    {
+        return _climbFeedbackLeft.atSetpoint();
+    }
+    
+    public boolean isAtRightSetpoint()
+    {
+        return _climbFeedbackRight.atSetpoint();
+    }
+
+    public void setExtensionMax(double max)
+    {
+        _climbMaxExtension = max;
+    }
+
+    public void setExtensionMin(double min)
+    {
+        _climbMinExtension = min;
+    }
+    
+    // Mainly for dashboard
+    public double getExtensionLeft()
+    {
+        return _inputs.extensionLeft;
+    }
+    
+    // Mainly for dashboard
+    public double getExtensionRight()
+    {
+        return _inputs.extensionRight;
+    }
+
+    public void algorithmStop()
+    {
+        _io.setAlgorithmVoltageLeft(0.0);
+        _io.setAlgorithmVoltageRight(0.0);
+    }
+    
+    public void setAlgorithmSetpointLeft(double leftHeight)
+    {
+        _leftHeight = leftHeight;
+    }
+    
+    public void setAlgorithmSetpointRight(double rightHeight)
+    {
+        _rightHeight = rightHeight;
+    }
+    
+    public void setAlgorithmVoltageLeft(double volts)
+    {
+        _leftHeight = null;
+        
+        _io.setAlgorithmVoltageLeft(volts);
+    }
+    
+    public void setAlgorithmVoltageRight(double volts)
+    {
+        _leftHeight = null;
+        
+        _io.setAlgorithmVoltageRight(volts);
+    }
+
+    public boolean isAtLeftSetpointAlgorithm()
+    {
+        return _leftPID.atSetpoint();
+    }
+
+    public boolean isAtRightSetpointAlgorithm()
+    {
+        return _rightPID.atSetpoint();
+    }
+
+    public void setHeight(double x)
+    {
+        _desiredHeight = x;
+    }
+    
+    public double getCurrentTilt()
+    {
+        return _gyro.getAngle();
```

Comment:
This will end up being replaced with a function call to the Gyro subsystem we'll create that's something like `getCurrentTilt`, but the method we need to call on the `AHRS` object won't be `getAngle`. `getAngle` measures the robot's yaw (left-to-right when viewed from the top) rotation, but we'll need to call the method that gets the robot's roll (left-to-right when viewed from behind) rotation.